serialport.go

```diff
@@ -287,6 +226,46 @@ func (p *serport) writerNoBuf() {
 	spList(false)
 }
 
+// this method runs as its own thread because it's instantiated
+// as a "go" method. so if it blocks inside, it is ok
+func (p *serport) writerRaw() {
+	// this method can panic if user closes serial port and something is
+	// in BlockUntilReady() and then a send occurs on p.sendNoBuf
+
+	defer func() {
+		if e := recover(); e != nil {
+			log.Println("Got panic: ", e)
+		}
+	}()
+
+	// this for loop blocks on p.sendRaw until that channel
+	// sees something come in
+	for data := range p.sendRaw {
+
+		// Decode stuff
+		sDec, err := base64.StdEncoding.DecodeString(data)
+		if err != nil {
+			log.Println("Decoding error:", err)
+		}
+		log.Println(string(sDec))
+
+		// we want to block here if we are being asked to pause.
+		goodToGo, _ := p.bufferwatcher.BlockUntilReady(string(data), "")
```

Comment:
Why ignoring the error?

Reply:
We saw that `p.bufferwatcher.BlockUntilReady` is just a placeholder, and the second parameter is not an error but a `bool`.